### PR TITLE
Fix log to file Windows tests

### PR DESCRIPTION
**Pull request by yinweisu**
*Issue #, if available:*

*Description of changes:*
* tempfile behaves weird on Windows: if you create a file within the TemporaryDirectory, the directory will fail to clean up itself. Minimal reproducible example
```python
def test_foo():
    with tempfile.TemporaryDirectory() as tempdir:
        temp = os.path.join(tempdir, "abc")
        with open(temp, "r") as f:
            pass
```
pytest running test above will fail^^^
* Removed usage of tempfile

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tabular/src/autogluon/tabular/predictor/predictor.py [hunk 1]
```diff
@@ -3315,6 +3315,7 @@ def load_log(cls, predictor_path: str = None, log_file_path: Optional[str] = Non
             assert predictor_path is not None, "Please either provide `predictor_path` or `log_file_path` to load the log file"
             file_path = os.path.join(predictor_path, "logs", cls._predictor_log_file_name)
         assert os.path.isfile(file_path), f"Log file does not exist at {file_path}"
+        lines = []
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/predictor/predictor.py:3318]: why is this necessary?

**Review by Innixma — APPROVED** [review]
LGTM assuming tests pass, had 1 comment above

File: tabular/src/autogluon/tabular/predictor/predictor.py (hunk 1 above)
Comment (yinweisu) [reply tabular/src/autogluon/tabular/predictor/predictor.py:3318]: Just found that variable is unbounded, so added a default value. Though it probably doesn’t matter cause it will not be assigned a value only when we failed to read the file, and that would raise an error already
